content/300-guides/500-other/880-troubleshooting-orm/100-help-articles/400-nextjs-prisma-client-dev-practices.mdx

```diff
@@ -23,21 +23,19 @@ The solution in this case is to instantiate a single instance `PrismaClient` and
 ```ts file=db.ts
 import { PrismaClient } from '@prisma/client'
 
-const prismaClientSingleton = () => {
-  return new PrismaClient()
+declare global {
+  var prisma: PrismaClient | undefined
 }
 
-type PrismaClientSingleton = ReturnType<typeof prismaClientSingleton>
-
-const globalForPrisma = globalThis as unknown as {
-  prisma: PrismaClientSingleton | undefined
```

Comment:
👋

Could you clarify why you removed this snippet?

This was left in here for when you're working with Prisma Client extensions and would like to infer the type of the Prisma Client instance. Let us know if there are any improvements we could make or how we can make it clearer 🙂

Reply:
Aliasing `globalThis` to a new variable and casting it (twice) seems more confusing, then using  `globalThis` directly. 
Feel free to close the MR if you disagree